specification/ml/_types/DiscoveryNode.ts

```diff
@@ -17,14 +17,32 @@
  * under the License.
  */
 
-import { Dictionary } from '@spec_utils/Dictionary'
-import { Id, Name } from '@_types/common'
+import { Dictionary, SingleKeyDictionary } from '@spec_utils/Dictionary'
+import { Id, Name, VersionString } from '@_types/common'
 import { TransportAddress } from '@_types/Networking'
+import { integer } from '@_types/Numeric'
 
-export class DiscoveryNode {
-  attributes: Dictionary<string, string>
+export type DiscoveryNode = SingleKeyDictionary<Id, DiscoveryNodeContent>
+
+export class DiscoveryNodeContent {
+  name?: Name
   ephemeral_id: Id
-  id: Id
+  transport_address: TransportAddress
+  external_id?: string
+  attributes?: Dictionary<string, string>
+  roles?: string[]
+  version?: VersionString
+  min_index_version?: integer
+  max_index_version?: integer
+}
+
+/**
+ * Alternative representation of DiscoveryNode used in ml.get_job_stats and ml.get_datafeed_stats
+ */
+export class DiscoveryNodeCompact {
   name: Name
+  ephemeral_id: Id
+  id: Id
   transport_address: TransportAddress
+  attributes?: Dictionary<string, string>
```

Comment:
Why would attributes be optional? In recorded examples, it's always set. But I don't understand where that code comes from.